test/all.js

```diff
@@ -0,0 +1,80 @@
+var varstruct = require('../')
+var test = require('tape').test
+
+var fixtures = [
+  {
+    description: 'Array',
+    type: varstruct.Array(2, varstruct.String(3)),
+    length: 6,
+    values: [
+      [['foo', 'bar'], '666f6f626172']
+    ],
+    invalid: {
+      encode: [
+        ['foobar', /^TypeError: Expected Array, got String foobar$/],
+        [1, /^TypeError: Expected Array, got Number 1$/],
+        [Buffer.alloc(3), /^TypeError: Expected Array, got Buffer/],
+        [['foo', 'bar', 'baz'], /^TypeError: Expected Array(Length: 2), got Array(Length: 3)/],
+        [['foo'], /^TypeError: Expected Array(Length: 2), got Array(Length: 1)/],
+        [undefined, /^TypeError: Expected Array, got undefined/]
```

Comment:
@fanatid this PR is still WIP,  so any real attention to be paid is that you are OK with these error messages... they use the `typeforce` format,  and I was most likely just going to use it verbatim.